frontends/onnx-frontend/third_party/patches/OnnxMlirConstProp.patch

```diff
@@ -0,0 +1,21 @@
+diff --git a/src/Transform/ONNX/ConstProp.cpp b/src/Transform/ONNX/ConstProp.cpp
+index 116a41d6..6a16f796 100644
+--- a/src/Transform/ONNX/ConstProp.cpp
++++ b/src/Transform/ONNX/ConstProp.cpp
+@@ -123,9 +123,14 @@ Value createReplacingConstantOp(
+ template <typename T>
+ using EnableNotBool = std::enable_if_t<!std::is_same_v<T, bool>>;
+ 
+-/// Checks whether a variadic value is produced by dense ONNXConstantOps.
++/// Checks whether a variadic value is produced by dense ONNXConstantOps and has
++/// same element type.
+ bool isVariadicOperandFromDenseONNXConstantOp(ValueRange operands) {
```

Comment:
maybe we don't need to check if they have the same data type? If their datatype mismatches, it will fail at op verification anyway

Reply:
> maybe we don't need to check if they have the same data type? If their datatype mismatches, it will fail at op verification anyway

Agree that, this check is redundant after fixing the incompatible operand types of concat